common/src/main/java/com/wynntils/features/wynntils/TelemetryFeature.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.features.wynntils;
+
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.core.components.Managers;
+import com.wynntils.core.config.Category;
+import com.wynntils.core.config.Config;
+import com.wynntils.core.config.ConfigCategory;
+import com.wynntils.core.features.UserFeature;
+import com.wynntils.core.mod.event.WynntilsCrashEvent;
+import com.wynntils.core.net.ApiResponse;
+import com.wynntils.core.net.UrlId;
+import com.wynntils.utils.JsonUtils;
+import java.util.Locale;
+import java.util.Map;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import org.apache.commons.lang3.exception.ExceptionUtils;
+
+@ConfigCategory(Category.WYNNTILS)
+public class TelemetryFeature extends UserFeature {
+    @Config
+    public boolean crashReports = true;
+
+    @SubscribeEvent
+    public void onCrash(WynntilsCrashEvent event) {
+        String title = "Crashed " + event.getType().toString().toLowerCase(Locale.ROOT) + ": " + event.getName() + "\n";
+        String trace = ExceptionUtils.getStackTrace(event.getThrowable());
+
+        ApiResponse apiResponse =
+                Managers.Net.callApi(UrlId.API_ATHENA_TELEMETRY_CRASH, Map.of("trace", title + trace));
+
+        apiResponse.handleJsonObject(json -> {
+            String response = JsonUtils.getNullableJsonString(json, "message");
+            String hash = JsonUtils.getNullableJsonString(json, "hash");
+            if (response.equals("Crash report logged successfully.")) {
```

Comment:
Qodana is right here. This is the only "outstanding" `getNullableJson*` method. Maybe let's change it?

Reply:
Maybe we should, yes, but that is a bit scary. I am using it to return null in the Wynncraft gear API json, so that needs to be rewritten. I do not want to fix that in this PR.